kumascript/src/live-sample.js

```diff
@@ -127,11 +127,18 @@ function getLiveSampleIDs(slug, source) {
   // be different from the current slug, from which to extract the sample ID.
   let result = [];
   for (let token of tokens) {
-    if (
-      token.type === "MACRO" &&
-      normalizeMacroName(token.name) === "embedlivesample" &&
-      token.args.length
-    ) {
+    if (token.type !== "MACRO") continue;
+    const normalizedMacroName = normalizeMacroName(token.name);
+    if (normalizedMacroName === "inheritancediagram") {
+      // The InheritanceDiagram is a special macro that, unlike other EmbedLiveSample
+      // macros in that it itself renders the "EmbedLiveSample" with a set of
+      // hardcoded options. The only thing that is variable and comes from the
+      // the raw HTML is the size. So make an exception for this otherwise, the
+      // HTML element ID is lost and this function won't discovered that it is
+      // in fact a live sample here.
+      token.args = ["inheritance_diagram", ...token.args];
```

Comment:
I don't think we need this, but maybe I'm missing something?

Reply:
Some pages do this:
```
{{InheritanceDiagram(600, 400)}}
```
and some do just:
```
{{InheritanceDiagram}}
```
There's a default inside the macro but I guess we ought to respect the values coming from the content, if set.

Reply:
Yeah, I was thinking the same, that we shouldn't mess with the token since it reflects the use of `InheritanceDiagram` rather than `EmbedLiveSample`.

Reply:
@peterbe We should remove this line:
```js
token.args = ["inheritance_diagram", ...token.args];
```
At best, it doesn't matter (it's never used anyway), but at worst, it will mislead us into thinking that it does.

Reply:
I'm confused. Did I merge this prematurely?

Reply:
I approved it thinking you would remove this line, but I should have been more clear and just requested a change. I think when you started work the next day, you saw that it had been approved and merged it as is?